snapshot/src/test/java/io/zeebe/snapshots/broker/impl/ReceivedSnapshotTest.java

```diff
@@ -134,21 +141,21 @@ public void shouldReturnFalseOnConsumingChunkTwice() throws Exception {
     final var transientSnapshot = senderSnapshotStore.newTransientSnapshot(index, term, 1, 0).get();
     transientSnapshot.take(
         p -> takeSnapshot(p, List.of("file3", "file1", "file2"), List.of("content", "this", "is")));
-    final var persistedSnapshot = transientSnapshot.persist();
+    final var persistedSnapshot = transientSnapshot.persist().join();
 
     // when
     final var receivedSnapshot =
         receiverSnapshotStore.newReceivedSnapshot(persistedSnapshot.getId());
 
     try (final var snapshotChunkReader = persistedSnapshot.newChunkReader()) {
-      receivedSnapshot.apply(snapshotChunkReader.next());
+      receivedSnapshot.apply(snapshotChunkReader.next()).join();
     }
 
     try (final var snapshotChunkReader = persistedSnapshot.newChunkReader()) {
-      final var success = receivedSnapshot.apply(snapshotChunkReader.next());
+      final var success = receivedSnapshot.apply(snapshotChunkReader.next()).join();
 
-      // then
-      assertThat(success).isFalse();
+      // then - TODO: Semantics have changed
```

Comment:
I guess you missed this when changing the tests. If this test no longer makes sense, we can remove it

Reply:
Oops. Yes. I think it make sense to keep the test, but test that success == true.